salvo/src/lib/cmd_exec.py

```diff
@@ -35,18 +36,32 @@ def run_command(cmd: str, parameters: CommandParameters) -> str:
       command
   """
   output = ''
+  params = parameters._asdict()
+  tmpfile = tempfile.TemporaryFile(
+      mode='w+', dir=params['cwd'], prefix='cmd_output'
+  )
+
   try:
     log.debug(f"Executing command: [{cmd}] with args [{parameters._asdict()}]")
     cmd_array = shlex.split(cmd)
-    output = subprocess.check_output(
-        cmd_array, stderr=subprocess.STDOUT, **parameters._asdict())
 
-    if isinstance(output, bytes):
-      output = output.decode('utf-8').strip()
+    subprocess.check_call(cmd_array, stdout=tmpfile, stderr=tmpfile,
+                          **parameters._asdict())
 
-    log.debug(f"Returning output: [{output}]")
   except subprocess.CalledProcessError as process_error:
     log.error(f"Unable to execute [{cmd}]: {process_error}")
     raise
 
+  finally:
+    tmpfile.flush()
```

Comment:
Is using the temp instead of a string here an improvement? Looks like we have to do some number of operations to get this output now (flush, seek, read). What do we gain by switching this into a file instead of a string. Is this relevant to #90 which was talking about automatic cleanup? Looks like the original solution was already performing cleanup, since it wasn't writing to any files, while the current version seems to be leaving the temp file behind.

Or are we trying to achieve something other than what is listed in #90?

Reply:
The output from some benchmarks is rather large.  This seems to work more consistently in capturing the output.

I kept this in since it did alert me to a few TemporaryDirectory objects getting prematurely reaped while addressing https://github.com/envoyproxy/envoy-perf/issues/90.

I'd prefer to keep it.

Reply:
Thank you for helping me understand this. Since this qualifies as a tricky (or non-obvious) code, can we add a inline comment explaining the need for this construct? This will help future maintainers.

Reply:
Sure thing. Let me update the PR.